### PR TITLE
Confirmation modal

### PR DESCRIPTION
**Pull request by amaury901130**
#### Description:

* Confirmation matched target

---

#### Preview:

Screen Shot 2020-06-05 at 14 51 51

### CODE REVIEW

File: RSDemoProject/Chat/Views/ChatModalViewController.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  
+  var viewModel: ChatModalViewModel!
+  let bubbleBorderRadius: CGFloat = 39
+  let avatarBorderRadius: CGFloat = 22
+  @IBOutlet weak var userNameLabel: EdgeInsetLabel!
+  @IBOutlet weak var userAvatarImage: UIImageView!
+  @IBOutlet weak var rightBubbleView: UIView!
+  @IBOutlet weak var leftBubbleView: UIView!
+  @IBOutlet weak var happyFaceRight: UIImageView!
+  @IBOutlet weak var happyFaceLeft: UIImageView!
```
Comment (germanStabile) [inline RSDemoProject/Chat/Views/ChatModalViewController.swift:21]: include type as suffix

File: RSDemoProject/Chat/Views/ChatModalViewController.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    viewModel.delegate = self
```
Comment (germanStabile) [inline RSDemoProject/Chat/Views/ChatModalViewController.swift:25]: just setup views here and remove ChatViewModelState altogether

File: RSDemoProject/Common/Models/User.swift [hunk 3]
```diff
@@ -12,9 +12,9 @@ struct User: Codable {
   var id: Int
   var username: String?
   var email: String?
-  var provider: String
-  var uid: String
-  var allowPasswordChange = false
+  var provider: String?
+  var uid: String?
+  var allowPasswordChange: Bool? = false
```
Comment (germanStabile) [inline RSDemoProject/Common/Models/User.swift:17]: why optional if it has a default value?

File: RSDemoProject/Chat/ViewModels/ChatModalViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+enum ChatModalState {
+  case conversationLoaded
+  case none
+}
+
+class ChatModalViewModel {
+  var conversation: MatchConversation!
+  
+  var state: ChatModalState = .none {
```
Comment (glm4) [inline RSDemoProject/Chat/ViewModels/ChatModalViewModel.swift:23]: I don't think you need state nor delegate at all in this view model.

File: RSDemoProject/Networking/Responses/CreateTargetResponse.swift [hunk 5]
```diff
@@ -8,6 +8,14 @@
 
 import Foundation
 
-struct CreateTargetResponse: Codable {
-  //todo...
+class CreateTargetResponse: NSObject, Codable {
```
Comment (glm4) [inline RSDemoProject/Networking/Responses/CreateTargetResponse.swift:11]: Why not just `TargetMatch` as a model, naming this as a response makes you think it only has to do with some request but you are using it later for all kind of stuff.

File: RSDemoProject/Home/Views/CreateTargetViewController.swift [hunk 6]
```diff
@@ -150,8 +155,9 @@ extension CreateTargetViewController: CreateTargetDelegate {
   func didUpdateCreateTargetState() {
     switch viewModel.state {
     case .targetCreated:
-      //todo: back with the target response
-      dismiss(animated: true, completion: nil)
+      if let newTargetResponse = viewModel.targetCreatedResponse {
+        delegate?.didCreateTargetResponse(targetResponse: newTargetResponse)
```
Comment (glm4) [inline RSDemoProject/Home/Views/CreateTargetViewController.swift:159]: Maybe `didCreateTarget(withMatch: targetMatch)`
Naming is very important for giving context, 'creating a target response' is not really a thing.

File: RSDemoProject/Home/ViewModels/CreateTargetViewModel.swift [hunk 7]
```diff
@@ -24,7 +24,7 @@ protocol CreateTargetDelegate: class {
 class CreateTargetViewModel {
 
   let areaUnit = "m"
-  var targetCreatedResponse: CreateTargetResponse?
+  var targetCreatedResponse: TargetMatch?
```
Comment (glm4) [inline RSDemoProject/Home/ViewModels/CreateTargetViewModel.swift:27]: rename the variable also

File: RSDemoProject/Home/ViewModels/MainViewModel.swift [hunk 8]
```diff
@@ -59,6 +66,17 @@ class MainViewModel {
     }
   }
   
+  var newTargetResponse: TargetMatch? {
```
Comment (glm4) [inline RSDemoProject/Home/ViewModels/MainViewModel.swift:69]: variable name missmatch

File: RSDemoProject/Home/Views/CreateTargetViewController.swift [hunk 9]
```diff
@@ -150,8 +155,9 @@ extension CreateTargetViewController: CreateTargetDelegate {
   func didUpdateCreateTargetState() {
     switch viewModel.state {
     case .targetCreated:
-      //todo: back with the target response
-      dismiss(animated: true, completion: nil)
+      if let newTargetResponse = viewModel.targetCreatedResponse {
+        delegate?.didCreateTarget(targetResponse: newTargetResponse)
```
Comment (glm4) [inline RSDemoProject/Home/Views/CreateTargetViewController.swift:159]: `targetMatch`

**Pull request merged**